src/breakupModel/simulation/Collision.cpp

```diff
@@ -39,17 +39,59 @@ void Collision::generateFragments() {
     }
 
     //The fragment Count, respectively Equation 4
-    auto fragmentCount = static_cast<size_t>(0.1 * std::pow(mass, 0.75) * std::pow(_minimalCharacteristicLength, -1.71));
-
-    const std::string debrisName{sat1.getName() + " & " + sat2.getName() + "-Collision-Fragment"};
-    this->createFragments(fragmentCount, debrisName, sat1.getPosition());
+    auto fragmentCount = static_cast<size_t>(0.1 * std::pow(mass, 0.75) *
+                                             std::pow(_minimalCharacteristicLength, -1.71));
+    this->createFragments(fragmentCount, sat1.getPosition());
 }
 
 void Collision::characteristicLengthDistribution() {
     //The pdf for Collisions is: 0.0101914/(x^2.71)
     Breakup::characteristicLengthDistribution(-2.71);
 }
 
+void Collision::assignParent() {
+    //The names of the fragments for a given parent
+    const Satellite &bigSat = _input.at(0);
+    const Satellite &smallSat = _input.at(1);
+    auto debrisNameBigPtr = std::make_shared<const std::string>(bigSat.getName() + "-Collision-Fragment");
+    auto debrisNameSmallPtr = std::make_shared<const std::string>(smallSat.getName() + "-Collision-Fragment");
+
+    //First specify the total mass
+    double inputMass = bigSat.getMass() + smallSat.getMass();
+    double resultMass = 0;
+    auto satIt = _output.begin();
+    for (; satIt != _output.end() && resultMass < inputMass; ++satIt) {
+        resultMass += satIt->getMass();
+    }
+    //Erase potential "too-much" mass
+    resultMass = resultMass > inputMass ? resultMass - (satIt - 1)->getMass() : resultMass;
+    _output.erase(satIt, _output.end());
```

Comment:
This is an issue with the Method. The process of calculating the debris fragments is not mass conserving.

The python implementation therefore makes sure that the produced mass does not exceed the input mass.
compare line 248ff. (https://github.com/richardklima/Space_debris_removal_model/blob/master/breakup.py#L248)
```python
  idx = np.argmin(np.cumsum(m) < M)
  if idx:
      fragments = fragments[:idx]
```

Until now the simulation always produced less mass which is logic given the fact that we only calculate debris of a specific size and larger. So, this line is more like a final insurance just in case.

Reply:
Hmm, but the question remains if it is required at all. I think we actually should never exceed the input mass.

Life question: Final insurance required or not?

So it might be better to just print a waringing then and let the user decide instead of deleting fragments

Reply:
Hmm, but the question remains if it is required at all. I think we actually should never exceed the input mass.

Life question: Final insurance required or not?

So it might be better to just print a warning then and let the user decide instead of deleting fragments

Reply:
So, I've changed the behaviour:
The simulation produces a warning if the output contains more mass than the input.
The idea is that the user can than still remove "too much" mass if it is wished or not.

Reply:
Ok, so final iteration:

- The potential reduction of fragments now takes place in the A/M function (because why calculate further A/M values if the mass budget is already consumed, only to throw them away some steps later? Not good)
- I've added a warning which is printed in case of an mass budget exceeding, which surprisingly does happend if one execute the same scenario multiple times